### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by menaka809**
I created separate local branch called feature/login for login API and Implemented login API for backend and push into remote branch called feature/login.

### CODE REVIEW

File: pet-server/controllers/loginController.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    const result = await loginService.loginUser(email, password);
+    res.status(200).json(result);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ success: false, message: error.message });
+  }
+};
+
+module.exports = {
+  login,
```
Comment (nimi-kaushal) [inline pet-server/controllers/loginController.js]: const LoginService = {...
}

module.exports = LoginService ;

File: pet-server/repository/usersRepository.js
Comment (nimi-kaushal) [inline pet-server/repository/usersRepository.js:1]: rename to userRrepository.js

File: pet-server/repository/usersRepository.js [hunk 2]
```diff
@@ -9,6 +9,19 @@ const createUser = async (userData) => {
...
+    return user;
+  } catch (error) {
+    throw new Error(`Error fetching user by email: ${error.message} `);
+  }
+};
+
 module.exports = {
   createUser,
+  getUserByEmail,
 };
```
Comment (nimi-kaushal) [inline pet-server/repository/usersRepository.js]: const UserRrepository= {...
}

module.exports = UserRrepository;

File: pet-server/services/loginService.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    });
+    return { success: true, message: 'Login successful', token };
+  } catch (error) {
+    throw new Error(`Login failed: ${error.message}`);
+  }
+};
+
+module.exports = {
+  loginUser,
+};
```
Comment (nimi-kaushal) [inline pet-server/services/loginService.js]: module.exports = LoginService;

File: pet-server/utils/validation.js [hunk 4]
```diff
@@ -70,9 +70,29 @@ const validatePaySlipData = (paySlipData) => {
...
+  return schema.validate(data);
+};
+
 module.exports = {
   passwordComplexity,
   validateUserData,
   validateCompanyData,
   validatePaySlipData,
+  validateLoginData,
 };
```
Comment (nimi-kaushal) [inline pet-server/utils/validation.js:97]: module.exports = Validation;

File: pet-server/services/loginService.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    // Compare the provided password with the hashed password using bcrypt
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+    if (!isPasswordValid) {
+      throw new Error('Invalid password');
+    }
+
+    // Generate a JWT token for authentication
+    const token = jwt.sign({ userId: user.id }, secretKey, {
```
Comment (nimi-kaushal) [inline pet-server/services/loginService.js]: just return

**Review by nimi-kaushal — CHANGES_REQUESTED** [review]

File: pet-server/controllers/loginController.js (hunk 1 above)
Comment (menaka809) [reply pet-server/controllers/loginController.js]: done

File: pet-server/repository/usersRepository.js
Comment (menaka809) [reply pet-server/repository/usersRepository.js:1]: done

File: pet-server/repository/usersRepository.js (hunk 2 above)
Comment (menaka809) [reply pet-server/repository/usersRepository.js]: done

File: pet-server/services/loginService.js (hunk 3 above)
Comment (menaka809) [reply pet-server/services/loginService.js]: done

File: pet-server/utils/validation.js (hunk 4 above)
Comment (menaka809) [reply pet-server/utils/validation.js:97]: done

File: pet-server/services/loginService.js (hunk 5 above)
Comment (menaka809) [reply pet-server/services/loginService.js]: all the changes are push to the remote feature/login  branch

**Review by nimi-kaushal — APPROVED** [review]

**Pull request merged**